src/classes/container.ts

```diff
@@ -192,8 +190,12 @@ export class Container {
             if (this.retina.isRetina) {
                 area = area / ((this.canvas.pxratio) * 2);
             }
+
+            const particlesNumber = this.options.particles.number.value;
+            const density = this.options.particles.number.density.value_area;
+
             /* calc number of particles based on density area */
-            let nb_particles = area * this.options.particles.number.value / this.options.particles.number.density.value_area;
+            let nb_particles = area * particlesNumber / density;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Identifier 'nb_particles' is never reassigned; use 'const' instead of 'let'.](https://app.codacy.com/manual/ar3s/tsparticles/pullRequest?prid=5011230)